### PR TITLE
[TEP-0144] Add feature flag and doc placeholder

### PR DESCRIPTION
**Pull request by QuanZhang-William**
# Changes
Part of [#7270][#7270]. In [TEP-0144][tep-0144] we proposed a new `enum` field to support built-in param input validation.

This commit adds the feature flag `enable-param-enum` and documentation placeholder for the feature.

/kind feature

[#7270]: https://github.com/tektoncd/pipeline/issues/7270
[tep-0144]: https://github.com/tektoncd/community/blob/main/teps/0144-param-enum.md

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
[WIP] Add `enable-param-enum` feature flag to gate the use of `Param.Enum` API field
```

### CODE REVIEW

**Review by Yongxuanzhang — APPROVED** [review]
this lgtm

**Comment by QuanZhang-William** [comment]
Hi @pritidesai @tektoncd/core-maintainers . I saw your comment here around getting reviews from >1 organizations: https://github.com/tektoncd/pipeline/pull/7260#issuecomment-1777439959

I totally agree with the point, especially for PR involves major changes/concepts (like introducing a new CRD); and I believe we follow the rule strictly in TEP proposals.

Going forward, I was wondering for smaller implementation PRs like this, do you feel comfortable if reviews are from the same organization to favour productivity, if they have good domain knowledge?

**Comment by jerop** [comment]
> Going forward, I was wondering for smaller implementation PRs like this, do you feel comfortable if reviews are from the same organization to favour productivity, if they have good domain knowledge?

@QuanZhang-William I also had the same thoughts, raised it for discussion at the governance/community meeting today and opened https://github.com/tektoncd/community/pull/1097 for discussion and to clarify the policy, you're welcome to also chime in there

**Review by jerop — APPROVED** [review]

**Pull request merged**
